lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -806,4 +843,37 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
           documentation: {"text/markdown", info.doc}
         }
   end
+
+  defp beam_lib_chunks(bin, key) do
+    with {:ok, {_, [{^key, value}]}} <- :beam_lib.chunks(bin, [key]) do
+      {:ok, value}
+    end
+  end
+
+  defp cell_id_from_module(node, module) do
+    node
+    |> :erpc.call(fn -> module.module_info(:compile)[:source] end)
+    |> to_string()
+    |> String.split("#cell:", trim: true)
+    |> Enum.at(1)
+  end
+
+  defp keyfind(list, key) do
+    List.keyfind(list, key, 0) || :error
+  end
+
+  defp find_type_value({type_kind, value}, name, arity) when type_kind in [:type, :opaque] do
+    case value do
+      {^name, {_, line, _, _}, vars} when length(vars) == arity ->
+        {:ok, line}
+
+      {^name, {_, 0, _, [{_, line, _} | _]}, vars} when length(vars) == arity ->
```

Comment:
Is this matching on a line being zero? Can you provide an example of what this is supposed to match? We shouldn’t match on lines being zero, that’s specific to Elixir.

Reply:
```elixir
  @type t :: term()
  @type foo :: foo(:bar)
  @type foo(var) :: {var, t()}
```

When I try to match the position with `{^name, {_, line, _, _}, vars}` for the `foo/0` and `foo/1`, it returns that the type is at line 0, but I can use the second pattern matching to get which line it really is.

```elixir
foo/0: {:foo, {:user_type, 7, :foo, [{:atom, 0, :bar}]}, []}
foo/1: {:foo, {:type, 0, :tuple, [{:var, 8, :var}, {:user_type, 8, :t, []}]}, [{:var, 8, :var}]}
```

Reply:
I see. It is probably best to not use Code.Typespec.fetch_types then and instead rely on `:abstract_code` as well. You will find entries like this in abstract_code:

```
       {:attribute, 29, :type,
        {:t,
         {:type, {29, 25}, :map,
```

So you still need to look inside to find the arity, but you won't have to guess the line. WDYT?

Reply:
`export_type` will be correct for Elixir code but not Erlang one. :(